api/authenticator.py

```diff
@@ -369,53 +372,33 @@ class LibraryAuthenticator(object):
 
     @classmethod
     def from_config(cls, _db, library):
-        """Initialize an Authenticator from site configuration.
+        """Initialize an Authenticator for the given Library based on its
+        configured ExternalIntegrations.
         """
-        
-        authentication_policy = Configuration.policy("authentication")
-        if not authentication_policy:
-            raise CannotLoadConfiguration(
-                "No authentication policy given."
-            )
-        
-        if (not isinstance(authentication_policy, dict)
-            or not 'providers' in authentication_policy):
-            raise CannotLoadConfiguration(
-                "Authentication policy must be a dictionary with key 'providers'."
-            )
-        bearer_token_signing_secret = authentication_policy.get(
-            'bearer_token_signing_secret'
+
+        bearer_token_signing_secret = OAuthAuthenticationProvider.bearer_token_signing_secret(
+            _db
```

Comment:
The bearer token signing secret is a site-wide configuration setting (currently the only one).